util/misc.go

```diff
@@ -53,3 +55,20 @@ func GetStack() []byte {
 	buf = buf[:stackSize]
 	return buf
 }
+
+// WithRecovery wraps goroutine startup call with recovery handle.
+// There are two out of box common handlers `WithPanicLogf` and `WithPanicMetricInc`,
+// Caller can custom define new handler use last optional parameters.
+func WithRecovery(ctx context.Context, fn func(ctx context.Context), handler func(r interface{})) {
+	defer func() {
+		r := recover()
+		if handler != nil {
+			handler(r)
+		}
+		if r != nil {
+			buf := GetStack()
+			log.Errorf("goroutine meet panic !!! result: %v, stack: %s", r, buf)
```

Comment:
maybe the following format is more readable:
```go
log.Errorf("panic in the recoverable goroutine: %v, stack trace:\n%s", r, buf)
```

for example, once paniced by "slice out of bounds", the log message is:
```
[ERRROR] panic in the recoverable goroutine: slice out of bounds, stack trace:
xxxx
xxxx
xxxx
[INFO] other logs...
```